### PR TITLE
Add + Update sources

### PR DESCRIPTION
**Pull request by dragonx943**
Add liliana parser + DocTruyen5s source

### CODE REVIEW

File: src/main/kotlin/org/koitharu/kotatsu/parsers/site/liliana/LilianaParser.kt [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+            )
+        }
+
+        override suspend fun getPages(chapter: MangaChapter): List<MangaPage> {
+            val fullUrl = chapter.url.toAbsoluteUrl(domain)
+            val doc = webClient.httpGet(fullUrl).parseHtml()
+            val script = doc.selectFirst("script:containsData(const CHAPTER_ID)")?.data()
+                ?: throw Exception("Failed to get chapter id")
+
+            val chapterId = script.substringAfter("const CHAPTER_ID = ").substringBefore(";")
```
Comment (Koitharu) [inline src/main/kotlin/org/koitharu/kotatsu/parsers/site/liliana/LilianaParser.kt:152]:
```suggestion
            val chapterId = script.substringAfter("const CHAPTER_ID = ").substringBefore(';')
```

File: src/main/kotlin/org/koitharu/kotatsu/parsers/site/liliana/LilianaParser.kt [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+
+            val ajaxUrl = buildString {
+                append("https://")
+                append(domain)
+                append("/ajax/image/list/chap/")
+                append(chapterId)
+            }
+
+            val ajaxResponse = webClient.httpGet(ajaxUrl)
+            val responseJson = JSONObject(ajaxResponse.requireBody().string())
```
Comment (Koitharu) [inline src/main/kotlin/org/koitharu/kotatsu/parsers/site/liliana/LilianaParser.kt:162]:
```suggestion
            val responseJson = webClient.httpGet(ajaxUrl).parseJson()
```
Do not operate with raw response if it is not required

File: src/main/kotlin/org/koitharu/kotatsu/parsers/site/liliana/LilianaParser.kt [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+
+            if (!responseJson.optBoolean("status", false)) {
+                throw Exception(responseJson.optString("msg"))
+            }
+
+            val pageListHtml = responseJson.getString("html")
+            val pageListDoc = org.jsoup.Jsoup.parse(pageListHtml)
```
Comment (Koitharu) [inline src/main/kotlin/org/koitharu/kotatsu/parsers/site/liliana/LilianaParser.kt:169]: import `Jsoup`

File: src/main/kotlin/org/koitharu/kotatsu/parsers/site/liliana/LilianaParser.kt [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+
+            return pageListDoc.select("div.separator[data-index]").map { page ->
+                val index = page.attr("data-index").toInt()
+                val url = page.selectFirst("a")?.attr("abs:href") ?: page.parseFailed("Image url not found")
```
Comment (Koitharu) [inline src/main/kotlin/org/koitharu/kotatsu/parsers/site/liliana/LilianaParser.kt:173]:
```suggestion
                val url = page.selectFirstOrThrow("a").attr("abs:href")
```

File: src/main/kotlin/org/koitharu/kotatsu/parsers/site/liliana/LilianaParser.kt [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+                MangaPage(
+                    id = generateUid(url),
+                    url = url,
+                    preview = null,
+                    source = source,
+                )
+            }.sortedBy { it.id.toInt() }
```
Comment (Koitharu) [inline src/main/kotlin/org/koitharu/kotatsu/parsers/site/liliana/LilianaParser.kt:180]: `id` is a random number based on the `url` seed, sorting by it is not a good idea

File: src/main/kotlin/org/koitharu/kotatsu/parsers/site/liliana/LilianaParser.kt [hunk 6]
```diff
@@ -0,0 +1,198 @@
...
+        }
+
+        protected open suspend fun getAvailableTags(): Set<MangaTag> = coroutineScope {
+            val doc = webClient.httpGet("https://$domain/filter").parseHtml()
+            doc.select("div.advanced-genres > div > .advance-item").mapNotNullToSet { element ->
```
Comment (Koitharu) [inline src/main/kotlin/org/koitharu/kotatsu/parsers/site/liliana/LilianaParser.kt:185]:
```suggestion
            doc.select("div.advanced-genres > div > .advance-item").mapToSet { element ->
```

File: src/main/kotlin/org/koitharu/kotatsu/parsers/site/liliana/LilianaParser.kt [hunk 7]
```diff
@@ -0,0 +1,198 @@
...
+                MangaTag(
+                    key = element.selectFirst("span")?.attr("data-genre") ?: return@mapNotNullToSet null,
+                    title = element.text().trim(),
```
Comment (Koitharu) [inline src/main/kotlin/org/koitharu/kotatsu/parsers/site/liliana/LilianaParser.kt:188]:
```suggestion
                    title = element.text().toTitleCase(sourceLocale),
```

File: src/main/kotlin/org/koitharu/kotatsu/parsers/site/liliana/LilianaParser.kt [hunk 8]
```diff
@@ -0,0 +1,198 @@
...
+                    source = source,
+                )
+            }.sortedBy { it.id.toInt() }
+        }
+
+        protected open suspend fun getAvailableTags(): Set<MangaTag> = coroutineScope {
+            val doc = webClient.httpGet("https://$domain/filter").parseHtml()
+            doc.select("div.advanced-genres > div > .advance-item").mapNotNullToSet { element ->
+                MangaTag(
+                    key = element.selectFirst("span")?.attr("data-genre") ?: return@mapNotNullToSet null,
```
Comment (Koitharu) [inline src/main/kotlin/org/koitharu/kotatsu/parsers/site/liliana/LilianaParser.kt:187]:
```suggestion
                    key = element.selectFirstOrThrow("span[data-genre]").attr("data-genre")
```

File: src/main/kotlin/org/koitharu/kotatsu/parsers/site/liliana/vi/DocTruyen5s.kt [hunk 9]
```diff
@@ -0,0 +1,15 @@
...
+import org.koitharu.kotatsu.parsers.MangaSourceParser
+import org.koitharu.kotatsu.parsers.model.MangaParserSource
+import org.koitharu.kotatsu.parsers.site.liliana.LilianaParser
+
+@MangaSourceParser("DOCTRUYEN5S", "DocTruyen5s", "vi")
+internal class DocTruyen5s(context: MangaLoaderContext) :
+    LilianaParser(context, MangaParserSource.DOCTRUYEN5S, "dongmoe.com") {
+
+    override suspend fun getAvailableTags() = super.getAvailableTags()
+    override suspend fun getFilterOptions() = super.getFilterOptions()
```
Comment (Koitharu) [inline src/main/kotlin/org/koitharu/kotatsu/parsers/site/liliana/vi/DocTruyen5s.kt:13]: Useless override

Comment (dragonx943) [reply src/main/kotlin/org/koitharu/kotatsu/parsers/site/liliana/vi/DocTruyen5s.kt]: :p

File: src/main/kotlin/org/koitharu/kotatsu/parsers/site/liliana/LilianaParser.kt (hunk 5 above)
Comment (dragonx943) [reply src/main/kotlin/org/koitharu/kotatsu/parsers/site/liliana/LilianaParser.kt]: removed

**Comment by dragonx943** [comment]
@Koitharu

**Comment by dragonx943** [comment]
Edit: Still have bugs when scrape images from DocTruyen5s

**Pull request merged**